query_builder_test.go

```diff
@@ -3,8 +3,158 @@ package qb
+	qbTable()
+}
+
+type testTable struct {
+	ID    string `dbtable:"users" db:"id" table:"foo" col:"foo_id"`
+	Name  string `db:"name" col:"foo_name"`
+	Email string `db:"email" col:"foo_email"`
+}
+
+func (t *testTable) qbTable() {}
+
+type testTableNoName struct {
+	ID    string `db:"id"`
+	Name  string `db:"name"`
+	Email string `db:"email"`
+}
+
+type testModel struct {
+	ID string `db:"id"`
+	TestModelWithTime
+}
+
+type TestModelWithTime struct {
+	CreatedAt time.Time `db:"created_at"`
+	DeletedAt time.Time `db:"deleted_at"`
+}
+
+type testModelType struct {
+	testModel `dbtable:"model"`
+	Name      string `db:"name"`
+	Email     string `db:"email"`
+}
+
+type testModelTypePtr struct {
+	*string
+	*testModel `dbtable:"model"`
+	Name       string `db:"name"`
+	Email      string `db:"email"`
+}
+
+func TestNew(t *testing.T) {
+	testTableInterface := func() testInterface {
+		return &testTable{}
+	}
+
+	s := "string"
+
+	type args struct {
+		i    interface{}
+		opts []Option
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *QueryBuilder
+		wantErr bool
+	}{
+		{"ok", args{testTable{}, nil}, &QueryBuilder{
+			Table:         "users",
+			Columns:       []string{"id", "name", "email"},
+			SelectDeleted: false,
+		}, false},
+		{"ok with interface", args{testTableInterface(), nil}, &QueryBuilder{
+			Table:         "users",
+			Columns:       []string{"id", "name", "email"},
+			SelectDeleted: false,
+		}, false},
+		{"ok with no name", args{testTableNoName{}, nil}, &QueryBuilder{
+			Table:         "test_table_no_name",
+			Columns:       []string{"id", "name", "email"},
+			SelectDeleted: false,
+		}, false},
+		{"ok with model", args{testModelType{}, nil}, &QueryBuilder{
+			Table:         "model",
+			Columns:       []string{"id", "created_at", "deleted_at", "name", "email"},
+			SelectDeleted: false,
+		}, false},
+		{"ok with model ptr", args{testModelTypePtr{string: &s}, nil}, &QueryBuilder{
+			Table:         "model",
+			Columns:       []string{"id", "created_at", "deleted_at", "name", "email"},
+			SelectDeleted: false,
+		}, false},
+		{"ok with table name", args{&testTable{}, []Option{WithTableName("mytable")}}, &QueryBuilder{
+			Table:         "mytable",
+			Columns:       []string{"id", "name", "email"},
+			SelectDeleted: false,
+		}, false},
+		{"ok with options", args{testTable{}, []Option{WithTableTag("table"), WithColumnTag("col")}}, &QueryBuilder{
+			Table:         "foo",
+			Columns:       []string{"foo_id", "foo_name", "foo_email"},
+			SelectDeleted: false,
+		}, false},
+		{"fail", args{"not a struct", nil}, nil, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := New(tt.args.i, tt.args.opts...)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("New() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("New() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestMust(t *testing.T) {
```

Comment:
I think you can easily merge what you're testing in `TestNew` & `TestMust` so that we only have 1 test.